projects/spectator/src/lib/spectator-host/create-factory.ts

```diff
@@ -52,10 +52,16 @@ export function createHostFactory<C, H = HostComponent>(typeOrOptions: Type<C> |
     jasmine.addMatchers(customMatchers as any);
     TestBed.configureTestingModule(moduleMetadata);
 
-    if (options.componentProviders.length || options.componentMocks.length) {
+    if (
```

Comment:
Could we extract it to a function and reuse it in both places, please?

Reply:
I have extracted to a function and also added the possibility to just override only providers or viewProviders. After more thought I have seen that my original code would always override both even if you specified overrides only for one of them.
I also merged the tests together into the component provider tests since both are connected.